File: DsAndAlgo/Python/src/Practice/util/list_utils.py
from src.Practice.ds.list_node import ListNode


class ListUtils:
    @staticmethod
    def create_list(arr: list) -> ListNode:
        """
        Create a list from an array.

        :param arr: List of elements to create the list from.
        :return: A new list containing the elements of the array.
        """
        if not arr:
            return None
        head = ListNode(arr[0])
        current = head
        for value in arr[1:]:
            current.next = ListNode(value)
            current = current.next
        return head

    @staticmethod
    def print_list(head: ListNode) -> None:
        """
        Print the elements of the list.

        :param head: The head of the list to print.
        """
        current = head
        while current:
            print(current.val, end=" -> ")
            current = current.next
        print("None")

File: DsAndAlgo/Java/Coding/src/main/java/org/practice/ListUtils.java
package org.practice;

public class ListUtils {
    public static ListNode createList(int[] arr) {
        if (arr == null || arr.length == 0) {
            return null;
        }
        ListNode head = new ListNode(arr[0]);
        ListNode current = head;
        for (int i = 1; i < arr.length; i++) {
            current.next = new ListNode(arr[i]);
            current = current.next;
        }
        return head;
    }

    public static void printList(ListNode head) {
        ListNode current = head;
        System.out.println("------------Linked List-------------");
        while (current != null) {
            System.out.print(current.val + " ");
            current = current.next;
        }
        System.out.println();
    }
    public static void printListVertical(ListNode head) {
        ListNode current = head;
        System.out.println("------------Linked List-------------");
        while (current != null) {
            System.out.println(current.val);
            current = current.next;
        }
        System.out.println();
    }
}
